components/scream/src/share/scream_session.cpp

```diff
@@ -0,0 +1,32 @@
+#include "scream_session.hpp"
+#include "share/util/scream_utils.hpp"
+
+#ifdef SCREAM_FPE
+# include <xmmintrin.h>
+#endif
+
+#include <Kokkos_Core.hpp>
+
+namespace scream {
+
+void activate_floating_point_exceptions_if_enabled () {
+#ifdef SCREAM_FPE
+  _MM_SET_EXCEPTION_MASK(_MM_GET_EXCEPTION_MASK() &
+                         ~( _MM_MASK_INVALID |
+                            _MM_MASK_DIV_ZERO |
+                            _MM_MASK_OVERFLOW |
+                            _MM_MASK_UNDERFLOW ));
```

Comment:
Might as well take out the UNDERFLOW one here while I'm thinking of it. We don't care about UNDERFLOW, and it will like happen a lot in some of the column physics.